src/EditModel/event-program/epics.js

```diff
@@ -403,37 +404,84 @@ const saveEventProgram = eventProgramStore
             .catch(err => Observable.of(saveEventProgramError(err))),
     );
 
+/**
+ * @param {Object} eventProgramStore d2-ui store
+ * @param {Object} d2 Instance of d2
+ * @return {Promise<eventProgramStore>} A promise that resolves eventProgramStore
+*/
+const checkProgramForRequiredValues = (eventProgramStore, d2) => {
```

Comment:
Is there a good reason this is not a simple predicate function? I think it complicates things making it a promise. 
I would've probably called it ```getMissingFieldsForProgram``` and return an  empty array if there are no missing fields, and just check for this inside the switchMap. Then you do not need the inner flatMap below either.

Reply:
Yeah that was my thought as well. But we'd need some kind of wrapper as the next function in the chain needs the data of the program as well, that's why I thought of an Either monad. The reason I went for a promise is that I didn't have to add functionality like a monad, but I could've used a simple object wrapper as well, I suppose..

Any arguments against me changing the return value to the following?
`{ eventProgramStore, missingFields }`

Reply:
@Birkbjo Please have a look at the changes I made to the epic, should be easier to understand now as well

Reply:
Neat, LGTM!